  <launch>

    <arg name="total_drones" value="5"/>
    <arg name="mode" value="3d"/> 

    <arg name="map_size_x" value="42.0"/>
    <arg name="map_size_y" value="30.0"/>
    <arg name="map_size_z" value="5.0"/>
    
    <arg name="odom_topic" value="visual_slam/odom"/>
    
    <arg name="origin_lat" value="30.0"/>
    <arg name="origin_lon" value="120.0"/>
    <arg name="origin_alt" value="0.0"/>

    <arg name="drone_id"    value="0"/>
    <arg name="init_x"      value="-20.0"/>
    <arg name="init_y"      value="0.0"/>
    <arg name="init_z"      value="1.2"/>
    <arg name="target_x"    value="20.0"/>
    <arg name="target_y"    value="0.0"/>
    <arg name="target_z"    value="1.0"/>

    <arg name="run_mode"    value="low"/>
    <!-- Launch Rviz node -->

    <include file="$(find ego_planner)/launch/rviz.launch"/>
    
    <include file="$(find co_navi_pkg)/launch/low_speed/map_drones_generator.launch">
        <arg name="map_size_x" value="$(arg map_size_x)"/>
        <arg name="map_size_y" value="$(arg map_size_y)"/>
        <arg name="map_size_z" value="$(arg map_size_z)"/>
    </include>

    <!-- drone x generator -->
    <!-- planner run in sim --> 
    <include file="$(find ego_planner)/launch/run_in_sim.launch">
        <arg name="drone_id"    value="$(arg drone_id)"/>
        <arg name="init_x"      value="$(arg init_x)"/>
        <arg name="init_y"      value="$(arg init_y)"/>
        <arg name="init_z"      value="$(arg init_z)"/>
        <arg name="target_x"    value="$(arg target_x)"/>
        <arg name="target_y"    value="$(arg target_y)"/>
        <arg name="target_z"    value="$(arg target_z)"/>

        <arg name="map_size_x"  value="$(arg map_size_x)"/>
        <arg name="map_size_y"  value="$(arg map_size_y)"/>
        <arg name="map_size_z"  value="$(arg map_size_z)"/>
        <arg name="odom_topic"  value="$(arg odom_topic)"/>
    </include>

    <!-- gps imu message generator -->
    <include file="$(find co_navi_pkg)/launch/methods/odom_2_gps_imu.launch">
        <arg name="drone_id"    value="$(arg drone_id)"/>
        <arg name="init_x"      value="$(arg init_x)"/>
        <arg name="init_y"      value="$(arg init_y)"/>
        <arg name="init_z"      value="$(arg init_z)"/>
    
        <arg name="origin_lat"  value="$(arg origin_lat)"/>
        <arg name="origin_lon"  value="$(arg origin_lon)"/>
        <arg name="origin_alt"  value="$(arg origin_alt)"/>
      </include>

    <!-- FUSION NODE IMU GPS-->
    <include file="$(find co_navi_pkg)/launch/methods/Fusions/fusion_IG.launch">
      <arg name="drone_id"    value="$(arg drone_id)"/>
      <arg name="init_x"      value="$(arg init_x)"/>
      <arg name="init_y"      value="$(arg init_y)"/>
      <arg name="init_z"      value="$(arg init_z)"/>
      <arg name="origin_lat"  value="$(arg origin_lat)"/>
      <arg name="origin_lon"  value="$(arg origin_lon)"/>
      <arg name="origin_alt"  value="$(arg origin_alt)"/>
      <arg name="run_mode"    value="$(arg run_mode)"/>
    </include>

    <include file="$(find co_navi_pkg)/launch/methods/Fusions/fusion_IV.launch">
      <arg name="drone_id"    value="$(arg drone_id)"/>
      <arg name="init_x"      value="$(arg init_x)"/>
      <arg name="init_y"      value="$(arg init_y)"/>
      <arg name="init_z"      value="$(arg init_z)"/>
      <arg name="run_mode"    value="$(arg run_mode)"/>
    </include>

    <node pkg="co_navi_pkg" type="error_analysis.py" name="error_analysis_$(arg drone_id)">
        <param name="drone_id" value="$(arg drone_id)"/>
        <param name="mode" value="$(arg run_mode)"/>
    </node>
    
      <!-- <node pkg = "co_navi_pkg" type="logs_save.py" name="log_save" >
        <param name="run_mode"    value="$(arg run_mode)"/>
        <param name="fusion_mode" value="imu_vision"/>
      </node> -->

      <include file="$(find co_navi_pkg)/launch/methods/logs_error_analysis.launch">
          <arg name="run_mode"    value="$(arg run_mode)"/>
          <arg name="fusion_mode" value="imu_vision"/>
      </include>
</launch>